packages/react/dialog/src/Dialog.tsx

```diff
@@ -123,37 +128,38 @@ type DialogOverlayPrimitive = Polymorphic.ForwardRefComponent<
 const DialogOverlay = React.forwardRef((props, forwardedRef) => {
   const { forceMount, ...overlayProps } = props;
   const context = useDialogContext(OVERLAY_NAME);
-  return (
+  return context.modal ? (
     <Presence present={forceMount || context.open}>
-      <DialogOverlayImpl data-state={getState(context.open)} {...overlayProps} ref={forwardedRef} />
+      <DialogOverlayImpl {...overlayProps} ref={forwardedRef} />
     </Presence>
-  );
+  ) : null;
 }) as DialogOverlayPrimitive;
 
+DialogOverlay.displayName = OVERLAY_NAME;
+
 type DialogOverlayImplOwnProps = Polymorphic.OwnProps<typeof Primitive>;
 type DialogOverlayImplPrimitive = Polymorphic.ForwardRefComponent<
   Polymorphic.IntrinsicElement<typeof Primitive>,
   DialogOverlayImplOwnProps
 >;
 
 const DialogOverlayImpl = React.forwardRef((props, forwardedRef) => {
+  const context = useDialogContext(OVERLAY_NAME);
   return (
     <Portal>
-      <Primitive {...props} ref={forwardedRef} />
+      <Primitive data-state={getState(context.open)} {...props} ref={forwardedRef} />
     </Portal>
   );
 }) as DialogOverlayImplPrimitive;
 
-DialogOverlay.displayName = OVERLAY_NAME;
-
 /* -------------------------------------------------------------------------------------------------
  * DialogContent
  * -----------------------------------------------------------------------------------------------*/
 
 const CONTENT_NAME = 'DialogContent';
 
 type DialogContentOwnProps = Polymorphic.Merge<
-  Polymorphic.OwnProps<typeof DialogContentImpl>,
+  Polymorphic.OwnProps<typeof DialogContentModal | typeof DialogContentNonModal>,
```

Comment:
Interesting, does this work? I think I usually do the following for this sort of thing: 

```jsx
type DialogContentOwnProps = Polymorphic.Merge<
	| Polymorphic.OwnProps<typeof DialogContentModal>
	| Polymorphic.OwnProps<typeof DialogContentNonModal>,
	{ ... }
> 
```

If yours works we can probably clean up a *bunch* of boilerplate elsewhere 😅

Reply:
😬 as far as I can tell, but I think they also both have the same API so I wouldn't know.
I guess we should try it on APIs were it does make a difference?

Reply:
My gut is now telling me that this is probably wrong 😀